homeassistant/helpers/__init__.py

```diff
@@ -98,14 +98,17 @@ def config_per_platform(config, domain, logger):
 
     while config_key in config:
```

Comment:
I'll make the changes you suggested for the config_for_platform stuff and split those into a separate pull request, since it's really not related to the scene component.  I think we have to be careful about the succinct format that you suggested, since a duplicate key will override previous entries (I think).  For example, with automation I have a few entries that would share the same platform key:
```yaml
automation:
  event:
    alias: Hall Light
    event_type: ...
  event:
    alias: Dim Hall Light
    event_type: ...
```
For switches, etc. it probably wouldn't be an issue, since it wouldn't make sense to have duplicates of a platform configured there.